### ISSUE TITLE
Migration script 20130227205845_add_state_to_context not idempotent

### ISSUE THREAD

**Issue by Werkov** [TracksApp/tracks#2400]
I upgraded from 2.2.3 to 2.3.0 and ran into some issues with the above mentioned script when I ran the db:migrate task (mostly because I needed to try the migration several times). Eventually, I (hopefully) successfully migrated the database.

When I upgraded from 2.3.0 to 2.4.1 the script again caused troubles because its changes were already deployed

```
/usr/local/tracks-2.4.1/db/migrate/20130227205845_add_state_to_context.rb:7:in
ActiveRecord::StatementInvalid: Mysql2::Error: Duplicate column name 'state': ALTER TABLE `contexts` ADD `state` varchar(20) DEFAULT 'active' NOT NULL
```

Because I knew the changes were executed from the previous migration (2.2.3 to 2.3.0), I disabled the script (renaming) during 2.3.0 to 2.4.1 migration.

I wonder if the particular migration script can be made idempotent (or if I broke its idempotence by modifying SQL table manually during the first migration). Thanks.

**Comment by ZeiP** [comment]
Not sure this is feasible: The problem is that the migration creates a new column and then populates it, after which it removes the old column. The migrations are meant to be run only once, it's the same for all the other migrations too.

That being said, I don't suppose there's any harm to doing this if someone submits a PR which works both when the migration hasn't been already ran and when it has, partially or fully. 

If you remember what kind of problem you ran into with the migration, we can anyway definitely try to fix the actual issue that you ran into while running the migration on the first time. Open a new issue for that if this is the case.
